lib/policyEvaluator/utils/conditions.js

```diff
@@ -19,130 +19,129 @@ conditions.findConditionKey = (key, requestContext) => {
+        // Value is the value of that header (ex. id of grantee)
+        's3:x-amz-grant-read': headers['x-amz-grant-read'],
+        's3:x-amz-grant-write': headers['x-amz-grant-write'],
+        's3:x-amz-grant-read-acp': headers['x-amz-grant-read-acp'],
+        's3:x-amz-grant-write-acp': headers['x-amz-grant-write-acp'],
+        's3:x-amz-grant-full-control': headers['x-amz-grant-full-control'],
+        // s3:x-amz-copy-source is x-amz-copy-source header if applicable on
+        // a put object
+        's3:x-amz-copy-source': headers['x-amz-copy-source'],
+        // s3:x-amz-metadata-directive is x-amz-metadata-directive header if
+        // applicable on a put object copy.  Determines whether metadata will
+        // be copied from original object or replaced. Values or "COPY" or
+        // "REPLACE".  Default is "COPY"
+        's3:x-amz-metadata-directive': headers['metadata-directive'],
+        // s3:x-amz-server-side-encryption -- Used to require that object put
+        // use server side encryption.  Value is the encryption algo such as
+        // "AES256"
+        's3:x-amz-server-side-encryption':
+            headers['x-amz-server-side-encryption'],
+        // s3:x-amz-storage-class -- x-amz-storage-class header value
+        // (STANDARD, etc.)
+        's3:x-amz-storage-class': headers['x-amz-storage-class'],
+        // s3:VersionId -- version id of object
+        's3:VersionId': headers['x-amz-version-id'],
+        // s3:LocationConstraint -- Used to restrict creation of bucket
+        // in certain region.  Only applicable for CreateBucket
+        's3:LocationConstraint': requestContext.getLocationConstraint(),
+        // s3:delimiter is delimiter for listing request
+        's3:delimiter': query.delimiter,
+        // s3:max-keys is max-keys for listing request
+        's3:max-keys': query['max-keys'],
+        // s3:prefix is prefix for listing request
+        's3:prefix': query.prefix,
+        // s3 auth v4 additional condition keys
+        // (See http://docs.aws.amazon.com/AmazonS3/latest/API/
+        // bucket-policy-s3-sigv4-conditions.html)
+        // s3:signatureversion -- Either "AWS" for v2 or
+        // "AWS4-HMAC-SHA256" for v4
+        's3:signatureversion': requestContext.getSignatureVersion(),
+        // s3:authType -- Method of authentication: either "REST-HEADER",
+        // "REST-QUERY-STRING" or "POST"
+        's3:authType': requestContext.getAuthType(),
+        // s3:signatureAge is the length of time, in milliseconds,
+        // that a signature is valid in an authenticated request.  So,
+        // can use this to limit the age to less than 7 days
+        's3:signatureAge': requestContext.getSignatureAge(),
+        // s3:x-amz-content-sha256 - Valid value is "UNSIGNED-PAYLOAD"
+        // so can use this in a deny policy to deny any requests that do not
+        // have a signed payload
+        's3:x-amz-content-sha256': headers['x-amz-content-sha256'],
+    }[key];
```

Comment:
I know this was here before, but creating a big object just to get a single field is an awful waste of resource. Since all checks are synchronous, we could keep that object and check it when we need to.